### ISSUE TITLE
npm-deps should support scoped packages

### ISSUE THREAD

**Issue by kurtharriger** [bhauman/lein-figwheel#587]
Configuration validator suggests that npm-deps key must be a keyword, however the reader doesn't like :@ so for scoped packages you must use a string instead:

```
                           :npm-deps {"@atlaskit/editor-cq" "4.3.1"}
```

Error at [:cljsbuild :builds 0 :compiler :npm-deps "@atlaskit/editor-cq" 0]
The key "@atlaskit/editor-cq" at (:cljsbuild :builds 0 :compiler :npm-deps) does not conform.
It should be a Keyword

**Comment by bhauman** [comment]
Did you try pretending a colon as in :"@asdf/asdf" ?

**Comment by bhauman** [comment]
Also, does scoped packages work in ClojureScript proper?

**Comment by kurtharriger** [comment]
:"@abc/def" didnt work either.  Was recently added/fixed in

https://dev.clojure.org/jira/browse/CLJS-2226

**Comment by swannodette** [comment]
I'll just add that `:npm-deps` keys do not need to be keywords. Strings are valid, and probably in general the better way to go.

**Comment by njj** [comment]
@swannodette Does anyone here have an example of requiring in a scoped package?  I'm having some issues with clojure thinking I'm trying to dereference because of the `@`.

**Comment by kurtharriger** [comment]
I created an issue several months ago

https://dev.clojure.org/jira/browse/CLJS-2451

There is a link to an repo where it can be reproduced by uncommenting the
require statement in core.cljs

I’ve since been using shadow-cljs but it doesn’t support babel transforms
to input dependencies and recently gave cljsbuild another try but didn’t
get much further than before.

I think the problem might be in the code the generates the foreign libs.
Perhaps it isn’t being registered correctly and the error is just
misleading.

I recently tried using the build api directly to generate the node-inputs.
https://clojurescript.org/guides/javascript-modules

I ran into some issues there in with the js script cljsbuild uses to
resolve deps. It had some issues resolving ./schema which was a folder with
index.js rather than a file also had issue with rxjs/Observable. I also had
this issue with rxjs in shadow-cljs initially and might be the similar root
cause

https://github.com/thheller/shadow-cljs/issues/149
